tendermint/tests/integration.rs

```diff
@@ -351,4 +349,75 @@ mod rpc {
         subs_client.close().await.unwrap();
         let _ = driver_handle.await.unwrap();
     }
+
+    async fn tx_search() {
+        let rpc_client = localhost_rpc_client();
+        let (mut subs_client, driver) =
+            WebSocketClient::new("tcp://127.0.0.1:26657".parse().unwrap())
+                .await
+                .unwrap();
+        let driver_handle = tokio::spawn(async move { driver.run().await });
+
+        let tx_info = broadcast_tx(
+            &rpc_client,
+            &mut subs_client,
+            "tx_search_key",
+            "tx_search_value",
+        )
+        .await
+        .unwrap();
+
+        let res = rpc_client
+            .tx_search(
+                Query::eq("tx.height", tx_info.height),
+                true,
+                1,
+                1,
+                Order::Ascending,
+            )
+            .await
+            .unwrap();
+        assert!(res.total_count > 0);
+        // We don't have more than 1 page of results
+        assert_eq!(res.total_count as usize, res.txs.len());
+        // Find our transaction
+        let txs = res
+            .txs
+            .iter()
+            .filter(|tx| tx.height.value() == (tx_info.height as u64))
+            .collect::<Vec<&ResultTx>>();
+        assert_eq!(1, txs.len());
+        assert_eq!(tx_info.tx, txs[0].tx.as_bytes());
+
+        subs_client.close().await.unwrap();
+        driver_handle.await.unwrap().unwrap();
+    }
+
+    async fn broadcast_tx(
+        http_client: &HttpClient,
+        websocket_client: &mut WebSocketClient,
+        key: &str,
```

Comment:
why not pass tx as a whole `broadcast_tx(http_client, websocket_client, tx)`?

Reply:
Ah, this is a decision from a previous iteration. I'll change this 👍